### PR TITLE
Launch ansible job with a specified zone.

### PR DESCRIPTION
**Pull request by lfu**
When calling launch_ansible_job with a zone specified in root object, [Ansible Tower template runner](https://github.com/ManageIQ/manageiq-providers-ansible_tower/blob/bc988c02332aabe0117ff7e167bb25efbbe8810f/app/models/manageiq/providers/ansible_tower/automation_manager/template_runner.rb#L1) would be called to put the request into the queue with specified zone. So the Ansible Tower job would be handled in the specified zone when it comes off of the queue.
 
Depends on https://github.com/ManageIQ/manageiq-providers-ansible_tower/pull/170.
Depends on https://github.com/ManageIQ/manageiq-automation_engine/pull/319.

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1623636

@miq-bot add_label enhancement, hammer/no

### CODE REVIEW

**Comment by gmcculloug** [comment]
@lfu Please review failing test.

**Comment by lfu** [comment]
@miq-bot add_label changelog/yes

**Comment by tinaafitz** [comment]
@mkanoor Please review.

File: content/automate/ManageIQ/AutomationManagement/AnsibleTower/Operations/StateMachines/Job.class/__methods__/launch_ansible_job.rb [hunk 1]
```diff
@@ -119,13 +120,30 @@ def extra_variables
                   ansible_vars_from_ws_values(result)
                 end
 
+                def zone_from_request
+                  @handle.root["zone"]
+                end
+
+                def create_job_in_zone(options)
+                  opts = options.merge(
+                    :name                => "Launcg Ansible job with template: #{job_template.name}",
```
Comment (tinaafitz) [inline content/automate/ManageIQ/AutomationManagement/AnsibleTower/Operations/StateMachines/Job.class/__methods__/launch_ansible_job.rb]: @lfu I think there's a typo for "Launch Ansible job..."

**Comment by lfu** [comment]
@bzwei @d-m-u Please review. Thanks.

**Comment by bzwei** [comment]
LGTM

**Pull request merged**